include/boost/graph/isomorphism.hpp

```diff
@@ -30,6 +30,22 @@ namespace boost
 namespace detail
 {
 
+    template < typename T >
+    struct HashableConcept {
+        BOOST_CONCEPT_USAGE(HashableConcept)
+        {
+            hash<T> hasher;
+            typedef typename hash<T>::result_type hash_result;
+            hash_result val = hasher(t);
+            boost::ignore_unused_variable_warning(val);
+        }
+
+        T t;
```

Comment:
Maybe I'm being too pedantic here, but declaring `T t;` here requires that `T` is `DefaultConstructible`, but is that necessary for the `Hashable` concept?

Reply:
No, that's a good catch, it's unnecessary. I have to admit I found this exact concept in some other area of boost and didn't look at it closely enough.

Reply:
I can't use `declval` in this context though to avoid implying a particular construction method. I can avoid it by refactoring the whole invariant concept checking with C++11 features and `boost/type_traits` like so:

```c++
        template < typename T >
        using TestHashable = decltype(hash<T>()(std::declval<T>()));

        template < typename T >
        struct Hashable: std::integral_constant<bool, boost::is_detected_v<TestHashable, T>> {};

        template < typename T >
        struct Invariant: std::integral_constant<bool, Hashable<T>::value && boost::has_less<T, T, bool>::value && boost::has_equal_to<T, T, bool>::value> {};
```

Downside is compiler error messages are probably worse than the boost concept checking variant. Assuming I didn't miss a way to avoid the default construction assumption, would you prefer I keep the `Hashable` concept as-is with default-construction or refactor it with `type_traits`?

Reply:
Ultimately, I think we shouldn't be defining a `Hashable` concept in an `isomorphism` algorithm file, we should be relying on an existing concept defined either in the standard library or a Boost concept library.
I think that can be done in the future as part of a broader modernization change to Boost.Graph, so for now, I'm thinking just take the Hashable concept stuff out, don't worry about it.

Reply:
And the CI tests are failing because `concept` is now a reserved word.  :)